src/components/Flow/hooks/useFlowParams.ts

```diff
@@ -6,24 +6,30 @@ export interface UseFlowParamsProps<TFlowContext> {
   requiredParams?: Array<keyof TFlowContext>
 }
 
+function hasRequiredParams<TParams extends object, TRequiredParams extends keyof TParams>(
+  params: TParams,
+  requiredParams: TRequiredParams[],
+): params is TParams & Required<Pick<TParams, TRequiredParams>> {
+  return requiredParams.every(param => param in params && typeof params[param] !== 'undefined')
+}
+
```

Comment:
This provides a type guard for the existing useFlowParams hook below. We can assert that all the required params are supplied as expected and then communicate that with TS. 

Without this TS determines a prop which is conditional in the flow interface (by design since it's not required for each flow component, only signature form) and throws an error

Reply:
Nice! Looks good! Great use of types here